satpy/modifiers/atmosphere.py

```diff
@@ -77,7 +77,9 @@ def __call__(self, projectables, optional_datasets=None, **info):
         projectables = projectables + (optional_datasets or [])
         if len(projectables) != 6:
             vis, red = self.match_data_arrays(projectables)
-            sata, satz, suna, sunz = get_angles(vis)
+            # Adjust the angle data precision to match the data
+            # This does not affect the accuracy visibly
+            sata, satz, suna, sunz = [d.astype(vis.dtype) for d in get_angles(vis)]
```

Comment:
We need to be careful with angles and geographical coordinates, because if they need to be interpolated for example, the conversion to and from cartesian coordinate needs float64 to be accurate enough.
But here I don't think it's a problem.

Reply:
Yes, shouldn't be a problem here. Rayleigh correction doesn't use cartesian coordinates as far as I can see. Also the original tests pass so there's that too.